debian/patches/0069-add-tonemap-videotoolbox.patch

```diff
@@ -0,0 +1,1867 @@
++        y2 = get_dithered_y(y2, d), y3 = get_dithered_y(y3, d);
++    }
++
++    float3 chroma_c = get_chroma_sample(c0, c1, c2, c3);
++    float3 chroma = lrgb2yuv(chroma_c);
++
++    dst1.write(float4(y0, 0.0f, 0.0f, 1.0f), uint2(x, y));
++    dst1.write(float4(y1, 0.0f, 0.0f, 1.0f), uint2(x + 1, y));
++    dst1.write(float4(y2, 0.0f, 0.0f, 1.0f), uint2(x, y + 1));
++    dst1.write(float4(y3, 0.0f, 0.0f, 1.0f), uint2(x + 1, y + 1));
++    if (is_non_semi_planar_out) {
++        dst2.write(float4(chroma.y, 0.0f, 0.0f, 1.0f), uint2(xi, yi));
++        dst3.write(float4(chroma.z, 0.0f, 0.0f, 1.0f), uint2(xi, yi));
++    } else {
++        dst2.write(float4(chroma.y, chroma.z, 0.0f, 1.0f), uint2(xi, yi));
++    }
++}
+Index: libavfilter/vf_tonemap_videotoolbox.m
+IDEA additional info:
+Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
+<+>UTF-8
+===================================================================
+diff --git a/libavfilter/vf_tonemap_videotoolbox.m b/libavfilter/vf_tonemap_videotoolbox.m
+new file mode 100644
+--- /dev/null	(revision bcb63879b775fef8085a5311f045dc1cc4960a98)
++++ b/libavfilter/vf_tonemap_videotoolbox.m	(revision bcb63879b775fef8085a5311f045dc1cc4960a98)
+@@ -0,0 +1,1106 @@
++/*
++ * Copyright (c) 2024 Gnattu OC <gnattuoc@me.com>
++ *
++ * This file is part of FFmpeg.
++ *
++ * FFmpeg is free software; you can redistribute it and/or
++ * modify it under the terms of the GNU Lesser General Public
++ * License as published by the Free Software Foundation; either
++ * version 2.1 of the License, or (at your option) any later version.
++ *
++ * FFmpeg is distributed in the hope that it will be useful,
++ * but WITHOUT ANY WARRANTY; without even the implied warranty of
++ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
++ * Lesser General Public License for more details.
++ *
++ * You should have received a copy of the GNU Lesser General Public
++ * License along with FFmpeg; if not, write to the Free Software
++ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
++ */
++
++#include <float.h>
++
++#include "libavutil/avassert.h"
++#include "libavutil/common.h"
++#include "libavutil/imgutils.h"
++#include "libavutil/opt.h"
++#include "libavutil/objc.h"
++#include "libavutil/hwcontext.h"
++
++#include "avfilter.h"
++#include "internal.h"
++#include "video.h"
++#include "colorspace.h"
++#include "dither_matrix.h"
++#include "metal/utils.h"
++
++#define params_cnt 8
++#define pivots_cnt (7+1)
++#define coeffs_cnt (8*4)
++#define mmr_cnt (8*6*4)
++#define params_sz params_cnt*sizeof(float)
++#define pivots_sz pivots_cnt*sizeof(float)
++#define coeffs_sz coeffs_cnt*sizeof(float)
++#define mmr_sz mmr_cnt*sizeof(float)
++
++extern char ff_vf_tonemap_videotoolbox_metallib_data[];
++extern unsigned int ff_vf_tonemap_videotoolbox_metallib_len;
++
++static const enum AVPixelFormat supported_formats[] = {
++    AV_PIX_FMT_NV12,
++    AV_PIX_FMT_P010,
```

Comment:
Missing format that matches `in_planes > 2` and `out_planes > 2`.
Is this intended? Or they are just not tested?

Reply:
These formats have not been tested, so I conservatively removed them. The code path for non-biplanar formats is kept just in case they can be used.